libraries/botframework-config/src/botConfiguration.ts

```diff
@@ -189,73 +265,168 @@ export class BotConfiguration extends BotConfigurationBase {
+                    }
+                    break;
+
+                case ServiceTypes.QnA:
+                    {
+                        let qnaService = <IQnAService>service;
+                        let command = `qnamaker export kb --kbId ${qnaService.kbId} --environment prod --subscriptionKey ${qnaService.subscriptionKey} --hostname ${qnaService.hostname} --endpointKey ${qnaService.endpointKey}`;
+                        if (progress) {
+                            progress(service, command, index, this.services.length);
+                        }
+                        let p = await exec(command);
+                        var json = p.stdout;
+                        // make sure it's json
+                        JSON.parse(json);
+                        await fsx.writeFile(folder + `/${qnaService.id}.qna`, json, { encoding: 'utf8' });
+                        recipe.resources.push({
+                            type: service.type,
+                            id: service.id,
+                            name: service.name
+                        });
+                    }
+                    break;
+
+                case ServiceTypes.Endpoint:
+                    if (progress) {
+                        progress(service, '', index, this.services.length);
+                    }
+                    recipe.resources.push(<IUrlResource>{
+                        type: ServiceTypes.Endpoint,
+                        id: service.id,
+                        name: service.name,
+                        url: (<IEndpointService>service).endpoint
+                    });
+                    break;
+
+                case ServiceTypes.BlobStorage:
+                    if (progress) {
+                        progress(service, '', index, this.services.length);
+                    }
+                    recipe.resources.push(<IBlobResource>{
+                        type: ServiceTypes.BlobStorage,
+                        id: service.id,
+                        name: service.name,
+                        container: (<IBlobStorageService>service).container
+                    });
+                    break;
+
+                case ServiceTypes.CosmosDB:
+                    if (progress) {
+                        progress(service, '', index, this.services.length);
+                    }
+                    recipe.resources.push(<ICosmosDBResource>{
+                        type: ServiceTypes.CosmosDB,
+                        id: service.id,
+                        name: service.name,
+                        database: (<ICosmosDBService>service).database,
+                        collection: (<ICosmosDBService>service).collection,
+                    });
+                    break;
+
+                case ServiceTypes.File:
+                    if (progress) {
+                        progress(service, '', index, this.services.length);
+                    }
+                    recipe.resources.push(<IFileResource>{
```

Comment:
Note that casting like this - while allowed - does't take advantage of typescript. We've had bugs in the tools repo from this. Please declare using the type, so the TSC compiler will provide checking on the parameters.